hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala

```diff
@@ -1,33 +1,71 @@
-import model.{CredentialReportDisplay, CronSchedule, HumanUser, AwsAccount => Account}
+import config.Config.getAnghammaradSNSTopicArn
+import model.{CronSchedule, VulnerableUser, AwsAccount => Account}
 import org.joda.time.Seconds
-import play.api.Logging
-import schedule.unrecognised.IamUnrecognisedUsers.{filterUnrecognisedIamUsers, getHumanUsers, getJanusUsernames}
+import play.api.{Configuration, Logging}
+import schedule.IamMessages.FormerStaff.disabledUsersMessage
+import schedule.IamMessages.disabledUsersSubject
+import schedule.Notifier.{notification, send}
+import schedule.unrecognised.IamUnrecognisedUsers.{filterUnrecognisedIamUsers, getJanusUsernames}
+import schedule.vulnerable.IamDisableAccessKeys.disableAccessKeys
+import schedule.vulnerable.IamRemovePassword.removePasswords
 import schedule.{CronSchedules, JobRunner}
 import services.CacheService
-import utils.attempt.FailedAttempt
+import utils.attempt.Attempt
 
-class IamUnrecognisedUserJob(cacheService: CacheService) extends JobRunner with Logging {
+import scala.concurrent.ExecutionContext
+
+class IamUnrecognisedUserJob(cacheService: CacheService, snsClient: AmazonSNSAsync, iamClients: AwsClients[AmazonIdentityManagementAsync], config: Configuration)(implicit val executionContext: ExecutionContext) extends JobRunner with Logging {
   override val id: String = "unrecognised-iam-users"
   override val description: String = "Check for and remove unrecognised human IAM users"
   override val cronSchedule: CronSchedule = CronSchedules.everyWeekDay
+  private val topicArn: Option[String] = getAnghammaradSNSTopicArn(config)
 
   def run(testMode: Boolean): Unit = {
     if (testMode) {
       logger.info(s"Skipping scheduled $id job as it is not enabled")
     } else {
       logger.info(s"Running scheduled job: $description")
     }
-  }
 
-  val dummyJanusData: JanusData = ???
-  val janusUsers: List[String] = getJanusUsernames(dummyJanusData)
-  val credsReport = cacheService.getAllCredentials
-  val iamHumanUsers: List[HumanUser] = getHumanUsers(credsReport)
-  val unrecognisedIamUsers: List[HumanUser] = filterUnrecognisedIamUsers(iamHumanUsers, "name", janusUsers)
+    val dummyJanusData = JanusData(
+      Set(AwsAccount("Deploy Tools", "deployTools")),
+      ACL(Map("firstName.secondName" -> Set.empty)),
+      ACL(Map.empty),
+      SupportACL(Map.empty, Set.empty, Seconds.ZERO),
+      None
+    )
+
+    val janusUsers: Seq[String] = getJanusUsernames(dummyJanusData)
+    val allCredsReports = cacheService.getAllCredentials
+
+    allCredsReports.map { case (account, eitherCredsReportOrFailure) =>
+      eitherCredsReportOrFailure.map { credsReport =>
+        val humanUsers = credsReport.humanUsers
+        val unrecognisedIamUsers: Seq[VulnerableUser] = filterUnrecognisedIamUsers(humanUsers, "name", janusUsers)
 
-  //TODO disable access key and remove password
-  //TODO send notification that this has been done
+        if(unrecognisedIamUsers.nonEmpty) {
```

Comment:
Question on scala best practice. What's the party line on side effects from a map? Should the  bit of code that runs when this is true be in a separate forEach?

Reply:
Party line is use foreach 👍 

There's probably a valuable refactor to give some of these methods return types and centralise the error handling, but I think that can come later and applies to the other scheduled job too